php/Terminus/Helpers/Input.php

```diff
@@ -195,22 +196,19 @@ public static function orgname($args, $key) {
    * @param [string] $label Prompt for STDOUT
    * @return [string] Site name
   */
-  public static function site(
-      $args = array(),
-      $key = 'site',
-      $label = 'Choose site'
-  ) {
-    // early return if a valid site has been offered
+  public static function sitename($args = array(), $key = 'site', $label = 'Choose site') {
+    // return early if sitename is provided in args
     if(isset($args[$key])) {
-      if($site = SiteFactory::instance($args[$key])) {
-        $site_name = $site->getName();
-        return $site_name;
-      }
+      return $args[$key];
     }
-    $sites   = SiteFactory::instance();
+    $sitesCache = new SitesCache();
+    $sitenames = array_map(function($site_cache) {
+      return $site_cache['name'];
+    }, $sitesCache->all());
```

Comment:
Yeah. I'm trying to be better with functional programming in PHP. Foreaching makes me drowsy.